File: 백준/Silver/10828. 스택/스택.cc
#include<iostream>
#include<stack>
using namespace std;

int main() {
	int n,m;
	stack<int>s;
	string str;
	cin >> n;
	for (int i = 0; i < n; i++) {
		cin >> str;
		if (str == "push") {
			cin >> m;
			s.push(m);
		}
		else if (str == "pop") {
			if (!s.empty()) {
				cout << s.top() << endl;
				s.pop();
			}
			else
				cout << -1 << endl;
		}
		else if (str == "empty") {
			if (s.empty())
				cout << 1 << endl;
			else
				cout << 0 << endl;
		}
		else if (str == "top") {
			if (s.empty()) {
				cout << -1 << endl;
			}
			else
				cout << s.top() << endl;
		}
		else if (str == "size") {
			cout << s.size() << endl;
		}

	}
	
}

File: 백준/Silver/10828. 스택/스택.java
import java.util.*;
import java.io.*;

public class Main{
	public static void main(String [] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringBuffer sb = new StringBuffer();
        Stack<Integer>stack = new Stack<>();
        
        int n = Integer.parseInt(br.readLine());
        String put;
        int num;
        for(int i=0; i<n; i++) {
        	StringTokenizer st = new StringTokenizer(br.readLine()," ");
        	put = st.nextToken();
        	
        	switch(put) {
        	case "push":
        		num = Integer.parseInt(st.nextToken());
        		stack.push(num);
        		break;
        	case "pop":
        		if (!stack.isEmpty()) {
                    sb.append(stack.pop()).append("\n");
                } else {
                    sb.append("-1\n");
                }
        		break;
        	case "top":
        		if(!stack.isEmpty()) {
        			sb.append(stack.peek()+"\n");
        		}
        		else
        			sb.append("-1\n");
        		
        		
        		break;
        	case "empty":
        		if(stack.isEmpty()) {
        			sb.append("1\n");
        		}
        		else
        			sb.append("0\n");
        		break;
        		
        	case "size":
        		sb.append(stack.size()+"\n");
        		
        		break;
        	
        	}
        }
        System.out.print(sb);
        
      
		
	}
}